### PR TITLE
Support for customizing web hook host

### PR DESCRIPTION
**Pull request by kishorenc**
This PR allows the customization of the web hook URL. This allows us to host Drone on an internal VPN and allow only the webhook URL to be exposed to the outside world.

We're running this fork successfully on our network - would love to get this merged upstream.

### CODE REVIEW

File: cmd/drone-server/server.go [hunk 1]
```diff
@@ -51,6 +51,12 @@ var flags = []cli.Flag{
 		Usage:  "server address",
 		Value:  ":8000",
 	},
+	cli.StringFlag{
+		EnvVar: "DRONE_HOOK_HOST",
+		Name:   "hook-host",
+		Usage:  "for generating webhook links",
```
Comment (tboerger) [inline cmd/drone-server/server.go]: Maybe something like override hook host? Currently it suggests that this must be set

**Comment by kishorenc** [comment]
@tboerger I've updated the description.

**Comment by laszlocph** [comment]
Hey guys, is there a plan to merge this? I for one would love to use it for the same reason @kishorenc created it. Thanks.

**Comment by jmccann** [comment]
I think this could help me with some testing I do.

**Comment by kishorenc** [comment]
@bradrydzewski These changes have been running fine for the past couple of months on our internal CI. Please consider merging this PR.

**Comment by iterion** [comment]
Just checking in, it would be really great to see this merged! If there is anything specific you want to see addressed we are more than happy to do so.

**Comment by 0rax** [comment]
Any news on this one ? I did just manually patch the latest release with the content of this PR and all seems to be working fine.

Our use case was also to be able to host our Drone inside an internal VPN with only hooks being exposed in our DMZ. It would be great if this feature can be officially supported.

Thanks

**Comment by leibowitz** [comment]
This seems to me as backward compatible and adding a feature people want. 
Pretty simple, please press the merge button. Almost one year to review this is not justifiable
Otherwise don't make the project open to external PRs and contributions

**Comment by bradrydzewski** [comment]
1.0 includes the following parameters to solve this issue:

```
DRONE_SERVER_PROXY_PROTO=https
DRONE_SERVER_PROXY_HOST=ec41d0de.ngrok.io
```

**Pull request closed without merging**